openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+if openlcs_dir not in sys.path:
+    sys.path.append(openlcs_dir)
+from libs.common import compress_source_to_tarball  # noqa: E402
+
+
+class SourceContainerHandler(object):
+    """
+    Object used for handle source in source containers.
+    """
+    def __init__(self, config=None):
+        self.config = config
+
+    @staticmethod
+    def get_component_flat(data, comp_type):
+        return {
+            'uuid': str(uuid.uuid4()),
+            'type': comp_type,
+            'name': data.get('name'),
+            'version': data.get('version'),
+            'release': data.get('release'),
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+
+    def get_container_component(self, sc_nvr):
+        """
+        Get container component.
+        """
+        nvr = koji.parse_NVR(sc_nvr)
+        container_component = self.get_component_flat(nvr, 'CONTAINER_IMAGE')
+        return container_component
+
+    def get_srpm_components(self, srpm_dir, sc_nvr):
+        """
+        Get the srpm components list from source container itself. The
+        component value is almost same as the srpm component from corgi.
+        The value of srpm component from corgi is as following:
+        {
+            'uuid': 'uuid',
+            'type': 'SRPM',
+            'name': 'libcom_err',
+            'version': '1.45.6',
+            'release': '2.el8',
+            'license': '',
+            'arch': '',
+            'synced': False
+        }
+        @params: srpm_dir, destination directory for srpm files' dir
+        """
+        components = []
+        if os.path.isdir(srpm_dir):
+            srpms = glob.glob(srpm_dir + "/*.src.rpm")
+            for srpm in srpms:
+                srpm = srpm.split('/')[-1]
+                rpm_nvr = srpm.replace('.src.rpm', '')
+                nvr = koji.parse_NVR(rpm_nvr)
+                component = self.get_component_flat(nvr, 'SRPM')
+                components.append(component)
+        else:
+            err_msg = f'Not found {srpm_dir}'
+            raise RuntimeError(err_msg) from None
+        return components
+
+    def get_source_of_srpm_component(self, srpm_dir, nvr):
+        """
+        Get the source file or srpm
+        """
+        srpm_file = nvr + '.src.rpm'
+        srpm_filepath = os.path.join(srpm_dir, srpm_file)
+        if os.path.isfile(srpm_filepath):
+            return srpm_filepath
+        else:
+            err_msg = f'Not found {srpm_filepath}'
+            raise RuntimeError(err_msg) from None
+
+    def get_misc_component(self, misc_dir, sc_nvr):
+        """
+        Get the metadata component of the source container.
+        After unpacking the source container, it has some matedata files.
+        The metadata files need to be gathered as one misc component.
+        @params: misc_dir, destination directory for source container metadata
+        @params: sc_nvr, source container nvr
+        """
+        components = []
```

Comment:
In order to consist with the other different types, or in the future the "UNKNOWN" type has more than one component.

Reply:
hmm. it is possible to have different types at all?

I think `misc` components like this will have problem syncing back to corgi - if each source container has a `misc` component, we'll never be able to have consistent container components between olcs/corgi.

IIRC we had a talk regarding this, and agreed to put all those metadata-stuff into the `container-component`, it is where those files resides. No?

Reply:
Updated according to the discussion in the daily sync meeting.